src/routes/safe/components/Apps/components/SafeAppsErrorBoundary.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, ErrorInfo } from 'react'
+import styled from 'styled-components'
+import { Text, Link, Icon, FixedIcon, Title } from '@gnosis.pm/safe-react-components'
+import { RouteComponentProps, withRouter } from 'react-router-dom'
+import { extractPrefixedSafeAddress, generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+
+type SafeAppsErrorBoundaryProps = {
+  children?: ReactNode
+} & RouteComponentProps
+
+type SafeAppsErrorBoundaryState = {
+  hasError: boolean
+  error?: Error
+}
+
+class SafeAppsErrorBoundary extends React.Component<SafeAppsErrorBoundaryProps, SafeAppsErrorBoundaryState> {
+  public state: SafeAppsErrorBoundaryState = {
+    hasError: false,
+  }
+
+  constructor(props: SafeAppsErrorBoundaryProps) {
+    super(props)
+    this.state = { hasError: false }
+
+    this.handleGoBack = this.handleGoBack.bind(this)
+  }
+
+  public componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
+    console.error('Uncaught error:', error, errorInfo)
+  }
+
+  public static getDerivedStateFromError(error: Error): SafeAppsErrorBoundaryState {
+    return { hasError: true, error }
+  }
+
+  private handleGoBack(): void {
+    this.props.history.push(generateSafeRoute(SAFE_ROUTES.APPS, extractPrefixedSafeAddress()))
```

Comment:
`extractPrefixedSafeAddress` can be potentially stale. We've pivoted to using the current `shortName` and `safeAddress` in the `currentSession` store entry.

Reply:
ok, changed